src/app/components/TextInput/TextInput.tsx

```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import type { ReactNode } from 'react';
+import styles from './TextInput.module.css';
+import Typography from '../Typography/Typography';
+
+type TextInputProps = {
+  inputType: 'text' | 'password';
+  children: ReactNode;
+};
+
+export default function TextInput({
+  inputType,
+  children,
+}: TextInputProps): JSX.Element {
+  return (
+    <div className={styles.inputField}>
+      <input type={inputType} required className={styles.input} />
+      <label className={styles.label}>
+        <span className={styles.labelContent}>
+          <Typography size="s">{children}</Typography>
+        </span>
+      </label>
```

Comment:
You need to connect the `input` element with the `label`. Best to add an `id` props to this component!
```suggestion
      <input id={id} type={inputType} required className={styles.input} />
      <label htmlFor={id} className={styles.label}>
        <span className={styles.labelContent}>
          <Typography size="s">{children}</Typography>
        </span>
      </label>
```

Reply:
that is a nice solution, I didn't even know the htmlFor property! Thanks for teaching me